components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
+import { AddModuleMutation, useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, MD_INPUT, TextField } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { ApolloError } from 'apollo-server-micro'
+
+type Props = {
+  title?: string
+  lessonId: number
+  onAddModule?: (
+    m: AddModuleMutation['addModule'] | null,
+    e: { name: string; content: string } | null
+  ) => void
+}
+
+enum Error {
+  InvalidData = 'missing module name or description'
+}
+
+const values: [TextField, TextField] = [
+  {
+    title: 'Module Name',
+    value: ''
+  },
+  {
+    title: 'Description',
+    type: MD_INPUT,
+    value: ''
+  }
+]
+
+const AdminLessonInputs = ({ title, lessonId, onAddModule }: Props) => {
+  const [formOptions, setFormOptions] = useState<any>(values)
+  const [name, content] = formOptions
+
+  const [addModuleMutation, { data, error, loading }] = useAddModuleMutation({
+    variables: {
+      content: content.value || '',
```

Comment:
Since we changed type to `TextField`, it should mean content.value guarantees to exist now as a string data type, so I think we can remove the `|| ''` since we don't have to check for undefined value anymore